src/borg/archive.py

```diff
@@ -1034,28 +1034,47 @@ def identify_key(self, repository):
         return key_factory(repository, cdata)
 
     def verify_data(self):
+        def chunker(it, size):
+            while True:
+                result = []
+                try:
+                    for i in range(size):
+                        result.append(next(it))
+                except StopIteration:
+                    yield result
+                    raise StopIteration
+                else:
+                    yield result
+
         logger.info('Starting cryptographic data integrity verification...')
         count = len(self.chunks)
         errors = 0
         defect_chunks = []
         pi = ProgressIndicatorPercent(total=count, msg="Verifying data %6.2f%%", step=0.01)
-        for chunk_id, (refcount, *_) in self.chunks.iteritems():
-            pi.show()
-            try:
-                encrypted_data = self.repository.get(chunk_id)
-            except Repository.ObjectNotFound:
-                self.error_found = True
-                errors += 1
-                logger.error('chunk %s not found', bin_to_hex(chunk_id))
-                continue
-            try:
-                _chunk_id = None if chunk_id == Manifest.MANIFEST_ID else chunk_id
-                _, data = self.key.decrypt(_chunk_id, encrypted_data)
-            except IntegrityError as integrity_error:
-                self.error_found = True
-                errors += 1
-                logger.error('chunk %s, integrity error: %s', bin_to_hex(chunk_id), integrity_error)
-                defect_chunks.append(chunk_id)
+        for chunk_infos in chunker(self.chunks.iteritems(), 100):
+            chunk_ids = [chunk_id for chunk_id, _ in chunk_infos]
+            chunk_ids_iter = iter(chunk_ids)
+            chunk_data_iter = self.repository.get_many(chunk_ids)
+            while True:
+                pi.show()
+                try:
+                    chunk_id = next(chunk_ids_iter)
+                    encrypted_data = next(chunk_data_iter)
```

Comment:
not sure here: if a exception like ObjectNotFound happens in this iterator, will it still be usable for next chunk?